docs/sop/PROVISIONING_PRODUCTION.md

```diff
@@ -0,0 +1,490 @@
+and run all the jobs in the staging build stage in your pipeline, and press "Run pipeline",
+then select`develop` and confirm   by clicking "Run pipeline".
+This will execute all automated jobs.
+There is a couple of manual jobs that will also need triggering: `PureFtpdBuildStaging` and then `PureFtpdDeployStaging`.
+
+
+When the manual and automated jobs have all completed successfully, it will result in a partial deployment of GigaDB website to the infrastructure we've just provisioned.
+Finally, you can perform the last step which is to load the environment's database server with data and install the tools on bastion servers needed by the users:
+
+```
+env OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES ansible-playbook -i ../../inventories data_cliapp_playbook.yml -e "gigadb_env=staging"
+```
+
+The website should be visitable at https://staging.gigadb.org, and the bastion server ready at bastion-stg.gigadb.host.
+
+## Staging for upstream/alt-gigab-website
+
+The provisioning for this project is similar to the other one, except that you need changing the key parameters that are
+described in the table in the "Upstream projects" section at the top of `docs/sop/DEPLOYING_TO_PRODUCTION.md`
+
+Change directory to the local checkout of the `upstream/alt-gigadb-website` project:
+
+```
+cd gigadb-alt-upstream
+```
+
+Change directory to the `envs` folder:
+```  
+cd ops/infrastructure/envs  
+```  
+
+Create directory `staging` directory if not existing already and change to it:
+```  
+mkdir staging
+```
+```  
+cd staging  
+```  
+
+Initialise Terraform, including creating a new state (or retrieving if existing) on Gitlab and copying terraform files to `staging` environment:
+```   
+../../../scripts/tf_init.sh --project gigascience/upstream/gigadb-website --env staging  
+ ```
+It's an interactive command and it will ask question you will need to answer like:
+  ```
+	You need to specify the path to the ssh private key to use to connect to the EC2 instance: </path/to/your-ssh-key-for-sydney-region>  
+      
+    You need to specify your GitLab username: <user input>  
+      
+    You need to specify a backup file created by the files-url-updater tool: </path/to/giagdbv3_*_v9.3.5.backup> (optional)  
+      
+    You need to specify an AWS region: ap-southeast-2  
+ ```  
+
+Alternatively you can specify most of the requested information as parameters to the commands:
+```  
+./../../scripts/tf_init.sh --project gigascience/upstream/gigadb-website --env staging --region ap-southeast-2 --ssh-key /path/to/your-ssh-key-for-sydney-region --web-ec2-type t3.medium --bastion-ec2-type t3.medium --rds-ec2-type "t3.large"  
+```  
+
+You can now  provision production staging server using the appropriate profile:
+```  
+AWS_PROFILE=UpstreamAlt terraform show   
+```
+>**Note**: will show all the existing resources, which should be the same as the terraform state file `staging_infra` from gitlab.
+
+```
+AWS_PROFILE=UpstreamAlt terraform plan
```

Comment:
**issue**: When executing this command, there are error messages:
```
 Error: no matching EC2 EIP found
│ 
│   with module.files_host.data.aws_eip.docker_host_eip,
│   on ../../modules/aws-instance/aws-instance.tf line 114, in data "aws_eip" "docker_host_eip":
│  114: data "aws_eip" "docker_host_eip" {
│ 
╵
╷
│ Error: no matching EC2 EIP found
│ 
│   with module.ec2_dockerhost.data.aws_eip.docker_host_eip,
│   on ../../modules/aws-instance/aws-instance.tf line 114, in data "aws_eip" "docker_host_eip":
│  114: data "aws_eip" "docker_host_eip" {
│ 
╵
╷
│ Error: no matching EC2 EIP found
│ 
│   with module.ec2_bastion.data.aws_eip.bastion_host_eip,
│   on ../../modules/bastion-aws-instance/bastion-aws-instance.tf line 84, in data "aws_eip" "bastion_host_eip":
│   84: data "aws_eip" "bastion_host_eip" {
```
**suggestion**: Updating the names of the 3 staging elastic IP addresses from `eip-gigadb-*-staging-gigadb` to `eip-gigadb*--staging-gigadbalt` resolves these errors.

Reply:
Hi @pli888 
thanks for spotting this.
I've fixed the documentation in commit [eb0e289](https://github.com/gigascience/gigadb-website/pull/1996/commits/eb0e289208aed208191b81312ce75535253f1d06),
let me know if you think it would have prevented you to create the situation that caused the error